### PR TITLE
GH-14885: [Docs] Make changes to the New Contrib Guide (Jira -> GitHub)

### LINKED ISSUE #14885: [Docs] Make changes to the New Contrib Guide (Jira -> GitHub)
### Describe the enhancement requested

Make updates to the New Contributors Guide due to change in the issue tracking workflow.

### Component(s)

Documentation

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR is part of the work to update the documentation as we are moving from Jira issue tracker to GitHub issues:
https://github.com/apache/arrow/issues/14822

Closes #14885
* Closes: #14885

### CODE REVIEW

**Comment by pitrou** [comment]
Hmm... can you give this a different title and description than #14919? I find this a bit confusing :-)

**Comment by AlenkaF** [comment]
> Hmm... can you give this a different title and description than #14919? I find this a bit confusing :-)

Oh 🤦‍♀️ Forgot to change the title of the PR :-)

File: docs/source/developers/guide/documentation.rst [hunk 1]
```diff
@@ -37,26 +37,15 @@ incomplete or inaccurate, share your hard-earned knowledge
...
-documentation itself, you can search for an issue on JIRA.
+documentation itself, you can search for an issue in GitHub.
 
 .. note::
-   When searching for JIRA issue that deals with documentation,
-   try selecting **Components** from the **More** tab in JIRA search
-   and select **Documentation** from the list.
+   When searching for an issue that deals with documentation,
+   navigate to `GitHub labels <https://github.com/apache/arrow/labels>`_
+   and select **Component:Documentation** or search for **Documentation**
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:45]:
```suggestion
   and select **Component: Documentation** or search for **Documentation**
```

File: docs/source/developers/guide/step_by_step/finding_issues.rst [hunk 2]
```diff
@@ -38,103 +38,50 @@ you might already have a bug to fix in mind, or a new feature that
...
 
-For both cases, JIRA is the issue tracker that we use.
+For both cases, GitHub is the issue tracker that we use.
 
-First we will explain how to use JIRA if you have a fix or a feature
-to work on that doesn't yet have a JIRA ticket open, in which case you
-will need to create a JIRA ticket yourself.
+- If you do not have a GitHub account yet, navigate to the
+  `GitHub login page <https://github.com/join>`_ to create one.
+- If you need help with creating a new GitHub issue see
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/step_by_step/finding_issues.rst:45]:
```suggestion
- If you need help with creating a new GitHub issue see the
```

File: docs/source/developers/guide/step_by_step/finding_issues.rst [hunk 3]
```diff
@@ -38,103 +38,50 @@ you might already have a bug to fix in mind, or a new feature that
...
+  `GitHub documentation <https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue>`_.
 
-Secondly, we will show you a way to find good first issues to work on.
+When the ticket is created you can start a discussion about it in theGitHub comments.
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/step_by_step/finding_issues.rst:48]:
```suggestion
When the ticket is created you can start a discussion about it in the GitHub comments.
```

File: docs/source/developers/guide/step_by_step/finding_issues.rst [hunk 4]
```diff
@@ -38,103 +38,50 @@ you might already have a bug to fix in mind, or a new feature that
...
 To make it easier for you to find issues that are well-suited for new
-contributors, we have added labels like “good-first-issue” or “beginner”
-to some JIRA tickets.
+contributors, we have added a label **“good-first-issue”** to some
+GitHub issues.
 
 .. seealso::
-   Search for good first/second issues with labels like in the `link here
-   <https://issues.apache.org/jira/issues/?filter=-4&jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20labels%20in%20(Beginner%2C%20beginner%2C%20beginners%2C%20beginnner%2C%20beginner-friendly%2C%20good-first-issue%2C%20good-second-issue%2C%20GoodForNewContributors%2C%20newbie%2C%20easyfix%2C%20documentation)%20order%20by%20created%20DESC>`_
+   Search for good first issues with labels like in the `link here
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/step_by_step/finding_issues.rst:58]:
```suggestion
   Search for good first issues `good-first-issue label listing
```

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks a lot for updating those guides! Looking good, added a few tiny comments

File: docs/source/developers/guide/index.rst [hunk 5]
```diff
@@ -128,7 +128,7 @@ of adding a basic feature.
 
    Communication is very important. You may need some help solving problems
    you encounter on the way (this happens to developers all the time). Also,
-   if you have a JIRA issue you want to solve, then it is advisable to let the team
+   if you have a GutHub issue you want to solve, then it is advisable to let the team
```
Comment (raulcd) [inline docs/source/developers/guide/index.rst:131]:
```suggestion
   if you have a GitHub issue you want to solve, then it is advisable to let the team
```

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**